### PR TITLE
Bump flint to 2.6.0

### PR DESCRIPTION
**Pull request by thofma**
Needs first the bump in LoadFlint.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -20,6 +20,6 @@ Antic_jll = "e21ec000-9f72-519e-ba6d-10061e575a27"
 AbstractAlgebra = "^0.9.1"
 BinaryProvider = "0.4, 0.5"
 julia = "1"
-LoadFlint = "^0.2.1"
-Arb_jll = "= 2.17.0"
-Antic_jll = "^0.1.0"
+LoadFlint = "= 0.2.1"
+Arb_jll = "= 2.18.0"
+Antic_jll = "= 0.2.1"
```
Comment (fingolfin) [inline Project.toml:25]: I am confused -- shouldn't this use LoadFlint 0.3.0 (which is not yet released) to get Flint 2.6.0 ?

**Comment by fingolfin** [comment]
Great, tests pass. So ideally this could be merged soon and then a new Nemo release be made. @thofma Is there anything speaking against that?

File: Project.toml [hunk 2]
```diff
@@ -20,6 +20,6 @@ Antic_jll = "e21ec000-9f72-519e-ba6d-10061e575a27"
 AbstractAlgebra = "^0.9.1"
 BinaryProvider = "0.4, 0.5"
 julia = "1"
-LoadFlint = "^0.2.1"
-Arb_jll = "= 2.17.0"
-Antic_jll = "^0.1.0"
+LoadFlint = "= 0.3.0"
+Arb_jll = "= 2.18.0"
+Antic_jll = "= 0.2.1"
```
Comment (fingolfin) [inline Project.toml:25]: Could you please explain what the rationale is for fixing the precise version here? In particular for the JLLs? Perhaps the reason could be inserted as a comment into the Project.toml?

**Comment by fingolfin** [comment]
I just noticed that this PR should probably also remove `build_FLINT.v0.0.1.jl`, `build_Antic.v0.2.1+1.jl`, `build_Arb.v2.17.0.jl` ?

File: Project.toml (hunk 2 above)
Comment (thofma) [reply Project.toml:25]: We probably don't need it on LoadFlint, but the other packages don't seem to following SemVer. So `FLINT_jll` version 2.7.0 or even 2.6.1 could in theory break Nemo that worked with version 2.6.0. So there is no safe upper bound that we could use.

**Comment by thofma** [comment]
I have to test if this breaks Hecke. If yes, we would want to include all the other breaking changes that we have accumulated.

**Comment by thofma** [comment]
And yes, I will remove the other files.

**Comment by thofma** [comment]
Hecke is failing with 2.6.0. We found a probably trivial bug with the new version, see https://github.com/wbhart/flint2/issues/783. So we need a fix and then an updated version of the binaries.

**Comment by wbhart** [comment]
At least @tthsqe12 has found the cause of that, as in the commit that causes it. But the commit is not in error, so we'll have to dig deeper to figure out why that causes the problem.

There are other things holding up 2.6.1 at the moment. But the debian people are helping and we are closing in hopefully. It's problematic that I will be on holidays again next week, so perhaps @tthsqe12 has to take over tracking down the bugs in Flint while I'm away.

**Comment by thofma** [comment]
This goes directly to 2.6.2/

Will merge as soon as I see lots of green.

**Pull request merged**